### ISSUE TITLE
OAuth Media API

### ISSUE THREAD

**Issue by spunkedy** [nwbb/node-brightcove#12]
Is there any current plans to have the media queries work under the new API?

**Comment by stevokk** [comment]
Are there any advantages to moving I guess is the question? If the old methods are being deprecated I haven't seen anything.

Cheers

**Comment by nwbb** [comment]
Agreed - I haven't become aware of anything being deprecated.  If it does, I guess we can asses then.

**Comment by spunkedy** [comment]
December 31st. I can find the exact link on their support site.

**Comment by stevokk** [comment]
Okay I have received this from BC support:

> On June 30, 2017, Brightcove will cease supporting the FTP Batch Provisioning System and the content ingest methods of the Media API. To add new content or to re-transcode existing titles in a Video Cloud account after June 30, 2017, all customers will need to use Dynamic Ingest. Introduced over 18 months ago, the Dynamic Ingest API supports all of the same formats and workflows as our legacy ingestion APIs and provides greater performance and scalability.  
> 
> Also on June 30, 2017, we will be retiring the write methods exposed by the Media API (e.g., update_video) and all customers and partners will need to migrate their applications to the CMS API prior to June 30, 2017. The CMS API has been designed to improve the performance and flexibility of Video Cloud integrations and works with all of the latest features of the new Brightcove Player and Video Cloud Studio.
> 
> All of the CMS connectors managed by Brightcove have been updated to use Dynamic Ingest, and content ingested via this method can be published with the new Brightcove Player or the legacy Smart Player. This migration guide details how custom integrations can be migrated to the new APIs.
> 
> On December 31, 2017, the “read” methods of the Media API will be fully retired. Customers that have built client-facing applications (e.g., web pages or mobile apps) that depend on the Media API must migrate to the Playback API before December 31, 2017. The Playback API has been designed for high-volume read operations and provides the functionality and scalability required for large scale apps.

**Comment by spunkedy** [comment]
Where did you find this? I can only find this:
https://support.brightcove.com/en/video-cloud/docs/migrating-new-video-cloud-studio-faq
```
When is the "end of life"  date for the legacy Studio?

We plan to close access to the legacy Video Cloud Studio on December 31, 2016. All users must be moved over to the new Studio by this date. Customers that need to maintain a legacy Smart Player after this date will be able to do so via special access to the legacy studio Players module, but legacy smart player creation will not be available. 

```

**Comment by stevokk** [comment]
That was from our support contact at Brightcove. I guess either a development branch or a new repo entirely needs to be started so a replacement for the old write api methods can be implemented before June, and read APIs shortly after.

@nwbb what are your thoughts on this? Are you actively using this library in production and willing to manage the new development?

From a personal position, I am soon to change off projects that involve brightcove so I won't be so active in development but I may continue to follow and help in my spare time.

**Comment by spunkedy** [comment]
I am capable of helping, but if there was a sample fork using the oauth and a quick use of the new read API. I might be able to help get it going quicker.

**Comment by nwbb** [comment]
Yeah, I've been thinking about it.  I am not using this in a production library, anymore, but many others are.  At this point, the differences in the API wouldn't mirror so well and I think this library should be sunsetted, with a pointer to the newer one.

**Comment by spunkedy** [comment]
I might be doing it before the end of the year, not sure yet. I have only a need for read of video / playlist data.

**Comment by nwbb** [comment]
That could be perfect timing.  I'd suggest borrowing whatever code you need from this in a new repo (rather than a fork), and I can update the readme to point to it.

**Comment by spunkedy** [comment]
The OAuth mechanism already works right? Once my time opens up to get this going I don't mind doing this.

**Comment by stevokk** [comment]
That'd be great start @spunkedy 

Hopefully during the deprecation period more developers will turn up looking for a new library and be willing to help. I know the developer taking over from me will face this same issue and I will drop him an email now explaining the situation.

Can I suggest that a notice is placed on the homepage now outlining the deprecation timeline and what they can do to help.

On the other hand, I don't see this library as over just yet, currently I am adding "update_image" to the library and it'd be good to get a final major version out and stable. Otherwise I'll be pointing at my fork for the foreseeable future.

**Comment by nwbb** [comment]
Agreed on all points.  I'll make the intention known in the readme and point to this issue for reference.  And @stevokk, I'll review your PR, merge it in, and update npm as necessary.

**Comment by nwbb** [comment]
Will also throw a message up to active consumers in npm via "deprecate".  Will be super useful once a new repo is created.

**Comment by spunkedy** [comment]
It will be at least a month before I can get to starting it.

**Comment by nwbb** [comment]
That's fine.  I'll just note the coming changes to the brighcove APIs that won't be reflected in this library.  If you do jump in with a new repo, just submit a pull request with a pointer to it in the readme and I'll get it up along with an updated message in npm.

**Comment by spunkedy** [comment]
Sounds great.

**Comment by nwbb** [comment]
Hey fellas - pushed the note up reflecting this issue.  @spunkedy if you do end up tossing up a successor, post the link to the repo here.

**Comment by spunkedy** [comment]
Here is the API client I started. It departed `slightly` from the previous paradigm. [nbright](https://github.com/spunkedy/nbright)

Any help generating more clients around it would be great. I will probably only be doing what I need for the project I am doing. 

If anyone has a test oauth access that we can use to properly run tests, I can put them into a build job with environment variables. ( READ only imo )

**Comment by spunkedy** [comment]
I can submit a PR if you want to update the main page.
